### ISSUE TITLE
OpenNTPd server does not respond to a READSTAT command.

### ISSUE THREAD

**Issue by cchildress** [openntpd-portable/openntpd-portable#37]
The nagios / icinga plugin [check_ntp_peer](https://www.monitoring-plugins.org/doc/man/check_ntp_peer.html) is intended to allow a monitoring server to confirm that a ntp server is healthy. It does this by sending a READSTAT opcode to the ntp server to request sync status and stratum info. I haven't been able to find any notes in the NTPv3 or NTPv4 RFC's that suggest this behavior has been deprecated.

It appears that OpenNTPd does not respond to this request. I've run a tcpdump from both the monitoring and the ntp servers and found that the ntp server does indeed receive the request, but does not respond.

Here is an example of the L4 segment and up of the outbound status request from one of my monitoring servers:
<pre>User Datagram Protocol, Src Port: 59515, Dst Port: 123
Network Time Protocol (NTP Version 2, control)
    Flags: 0x16, Leap Indicator: no warning, Version number: NTP Version 2, Mode: reserved for NTP control message
    Flags 2: 0x01, Response bit: Request, Opcode: READSTAT
    Sequence: 1
    Status: 0x0000
    AssociationID: 0
    Offset: 0
    Count: 0</pre>

**Comment by perkelix** [comment]
Is this still an issue? Can it be reproduced with 6.8p1 or newer?

**Comment by cchildress** [comment]
I'm afraid I don't have a good way to test anymore. I no longer maintain OpenNTPd servers and I don't have nagios handling health checks anymore...

The original bug is that OpenNTPd just reply to the READSTAT at all. My suggestion if you wanted to close this fully with a nice test case is to write a small helper test that sends that opcode. Then you have something in the repo here that can easily be used to quickly spin up a server and send this type of query.

**Comment by busterb** [comment]
Yeah, OpenNTPD is intentionally only implementing a subset of the NTP, and so READSTAT handling simply does not exist. A plugin like [check_ntp_time](https://nagios-plugins.org/doc/man/check_ntp_time.html) seems like it would work though, for anyone who is looking to solve this problem in the future.
